ledger/src/blockstore_db.rs

```diff
@@ -1574,27 +1568,20 @@ where
         result
     }
 
-    pub fn multi_get_bytes(
-        &self,
-        keys: impl Iterator<Item = C::Index>,
-    ) -> Vec<Result<Option<Vec<u8>>>> {
-        let rocks_keys: Vec<_> = keys.map(|key| C::key(key)).collect();
+    pub(crate) fn multi_get_bytes<I>(&self, keys: I) -> Vec<Result<Option<Vec<u8>>>>
+    where
+        I: IntoIterator<Item = C::Index>,
```

Comment:
An `IntoIterator` (instead of `Iterator`) allows to use any object which can be converted to an iterator (including iterators themselves). So it is the preferred type for the function arguments.